src/Popover/Popover.js

```diff
@@ -53,7 +79,11 @@ class Popover extends Component {
         }
 
         const referenceComponent = React.cloneElement(control, {
-            onClick: onClickFunctions
+            onClick: onClickFunctions,
+            tabIndex: 0,
+            role: 'button',
+            'aria-haspopup': true,
+            onKeyPress: (event) => this.handleKeyPress(event, control, onClickFunctions)
```

Comment:
I'm working on an additional fix for the popover target that adds `aria-haspopup='dialog' or 'grid' or 'listbox' or 'menu' or 'tree'` based on a new `type` property for the Popover component. `type` will default to `true`. This is to align with with [this spec](https://www.w3.org/TR/wai-aria-1.1/#aria-haspopup). It also sets `aria-expanded` to `true` or `false` based on state. It also sets appropriate value for `aria-controls`.

I will open a PR with those changes after this is merged as it essentially builds on top of your work.